demo/src/main/java/com/restaurant/demo/controller/ProductsController.java

```diff
@@ -9,26 +11,45 @@
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 import com.restaurant.demo.entity.Product;
+import com.restaurant.demo.exception.ApiRequestException;
 import com.restaurant.demo.service.ProductService;
 
 @RestController
 @RequestMapping("/products")
 public class ProductsController {
 	
 	@Autowired
-	ProductService productService;
+	private ProductService productService;
 	
 	@GetMapping()
 	public ArrayList<Product> listProducts(){
 		return productService.listProducts();
 	}
 	
+	@GetMapping(path = "/category")
```

Comment:
if you add this path, it means you're going to list the categories:
```
/api/products => it will list products
/api/products/1 => it will return the product with id 1
/api/products?category=pastas => it will return all products under category pastas
```

and you're doind this:
```
/api/products/category => I understand you're going to list the categories
```